Fix/fix error messages

JaneMoroz opened the pull request:
# Description

- created `ErrorComponent` 
   - in case of any server side error instead of `Error:...` `ErrorComponent` gets rendered. I also created `ErrorType` enum, which we can pass as an `errorType` prop in order for users (or us idk) to know more about the error and to get a more user friendly message.
   - also in case of any client side error this component also gets returned through the nearest `error.tsx`, but the msg for the user is gonna be the same no matter what kind of error occured: Something went wrong. Please contact support.
- added `error.tsx` to `/(main)`. Not sure about adding one to `/auth`.
- changed a message in `ErrorModal` to more user friendly one.
- about `getDashboardData`
   - I had to fix `fetchMeetingsPromises` logic a bit, because there were no error handling. Also a logic is a bit diferent there from other pages (probably because a lot of data need to be fetched) so had to add `errorMessage` and `errorType` props there and then in case of an error render `ErrorComponent` in `VoyageDashboard`

## Issue link

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration


# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

Dan-Y-Ko reviewed (changes requested):
Idk how I feel about how the message looks as is. What if we included the status code as well? Something like this https://i.imgur.com/I5KwXYh.png

JaneMoroz commented:
> Idk how I feel about how the message looks as is. What if we included the status code as well? Something like this https://i.imgur.com/I5KwXYh.png

I don't think we can get status codes 😓

Dan-Y-Ko commented:
> > Idk how I feel about how the message looks as is. What if we included the status code as well? Something like this https://i.imgur.com/I5KwXYh.png
> 
> I don't think we can get status codes 😓

We can, just need to change the requests.tsx file a bit. That screenshot I showed is from a change I made :)

JaneMoroz commented:
> > > Idk how I feel about how the message looks as is. What if we included the status code as well? Something like this https://i.imgur.com/I5KwXYh.png
> > 
> > 
> > I don't think we can get status codes 😓
> 
> We can, just need to change the requests.tsx file a bit. That screenshot I showed is from a change I made :)

Done! I also added `startTransition` in order to reset+refresh the page when `Try again` is clicked. I added these lines (`getCurrentSprint.ts` file) to test the logic (like something weird happened on server but then it fixed itself):
```
  const randomValue = Math.random();
  console.log(randomValue);
  const currentDate = randomValue > 0.5 ? new Date(2024, 5, 1) : new Date();
```
Without `startTransition` , the page actullly doesn't refresh even though it should.
